### PR TITLE
koord-manager: refactor config validation by github.com/go-playground

### PR DESCRIPTION
**Pull request by chzhj**
### Ⅰ. Describe what this PR does
see https://github.com/koordinator-sh/koordinator/issues/1232

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: apis/extension/slo_controller_config.go [hunk 1]
```diff
@@ -35,15 +35,15 @@ const (
 // +k8s:deepcopy-gen=true
 type NodeCfgProfile struct {
 	// like ID for different nodeSelector; it's useful for console so that we can modify nodeCfg or nodeSelector by name
-	Name string `json:"name,omitempty"`
+	Name string `json:"name,omitempty" validate:"required"`
```
Comment (saintube) [inline apis/extension/slo_controller_config.go:38]: This should be optional by default, and the check could be enabled via a command line argument.

File: apis/slo/v1alpha1/nodeslo_types.go [hunk 2]
```diff
@@ -229,26 +229,26 @@ type CPUBurstConfig struct {
...
 	// specifies a period of time for pod can use at burst, default = -1 (unlimited)
-	CFSQuotaBurstPeriodSeconds *int64 `json:"cfsQuotaBurstPeriodSeconds,omitempty"`
+	CFSQuotaBurstPeriodSeconds *int64 `json:"cfsQuotaBurstPeriodSeconds,omitempty" validate:"omitempty,min=-1"`
 }
 
 type CPUBurstStrategy struct {
 	CPUBurstConfig `json:",inline"`
 	// scale down cfs quota if node cpu overload, default = 50
-	SharePoolThresholdPercent *int64 `json:"sharePoolThresholdPercent,omitempty"`
+	SharePoolThresholdPercent *int64 `json:"sharePoolThresholdPercent,omitempty" validate:"omitempty,min=0"`
```
Comment (saintube) [inline apis/slo/v1alpha1/nodeslo_types.go:242]: `...,max=100`

File: apis/slo/v1alpha1/nodeslo_types.go [hunk 3]
```diff
@@ -229,26 +229,26 @@ type CPUBurstConfig struct {
 	// cpu burst percentage for setting cpu.cfs_burst_us, legal range: [0, 10000], default as 1000 (1000%)
 	// +kubebuilder:validation:Maximum=10000
 	// +kubebuilder:validation:Minimum=0
-	CPUBurstPercent *int64 `json:"cpuBurstPercent,omitempty"`
+	CPUBurstPercent *int64 `json:"cpuBurstPercent,omitempty" validate:"omitempty,min=1,max=10000"`
 	// pod cfs quota scale up ceil percentage, default = 300 (300%)
-	CFSQuotaBurstPercent *int64 `json:"cfsQuotaBurstPercent,omitempty"`
+	CFSQuotaBurstPercent *int64 `json:"cfsQuotaBurstPercent,omitempty" validate:"omitempty,min=1"`
```
Comment (saintube) [inline apis/slo/v1alpha1/nodeslo_types.go:234]: `...,min=100`

File: pkg/util/sloconfig/config.go [hunk 4]
```diff
@@ -18,14 +18,22 @@ package sloconfig
...
 	// SLOCtrlConfigMap is the name of the slo-controller configmap.
 	SLOCtrlConfigMap = "slo-controller-config"
+	// DefaultTranslator = "en"
+	DefaultTranslator = "en"
 )
 
 func InitFlags(fs *flag.FlagSet) {
 	fs.StringVar(&SLOCtrlConfigMap, "slo-config-name", SLOCtrlConfigMap, "determines the name the slo-controller configmap uses.")
 	fs.StringVar(&ConfigNameSpace, "config-namespace", ConfigNameSpace, "determines the namespace of configmap uses.")
+	fs.StringVar(&DefaultTranslator, "default_translator", DefaultTranslator, "determines the sloConfig validator translator.")
```
Comment (saintube) [inline pkg/util/sloconfig/config.go:38]: how about (`..., "slo-config-translator", DefaultTranslator, "determines the sloConfig validator translator. e.g. 'en', 'zh'")`?

File: apis/extension/slo_controller_config.go (hunk 1 above)
Comment (zwzhang0107) [inline apis/extension/slo_controller_config.go:38]: this field is an internal field managed by console, maybe we don't need to check it.

Comment (chzhj) [reply apis/extension/slo_controller_config.go:38]: enabled via a command line argument

File: pkg/util/sloconfig/validator.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+func GetValidatorInstance() *DefaultValidator {
+	validatorInstance.once.Do(func() {
+		validatorInstance.validator, validatorInstance.trans = createValidator()
+	})
+	return validatorInstance
+}
+
+func createValidator() (*validator.Validate, *ut.Translator) {
+	instance := validator.New()
+	switch DefaultTranslator {
```
Comment (eahydra) [inline pkg/util/sloconfig/validator.go:59]: I talked with @chzhj  and got some information. Supporting translator can make the error message more readable, but we need to consider whether it is necessary to translate the externally exposed verification error message from English to other languages in webhook support. cc @zwzhang0107 @saintube @hormes @FillZpp

Comment (zwzhang0107) [reply pkg/util/sloconfig/validator.go:59]: xhow about open another issue to discuss this; this PR only involves validation optimization.

**Review by eahydra — COMMENTED** [review]
/lgtm

**Review by hormes — APPROVED** [review]

**Pull request merged**